stan/math/opencl/matrix_cl.hpp

```diff
@@ -346,14 +346,72 @@ class matrix_cl<T, require_arithmetic_t<T>> {
     try {
       buffer_cl_ = cl::Buffer(ctx, CL_MEM_READ_WRITE, sizeof(T));
       cl::Event transfer_event;
-      queue.enqueueWriteBuffer(buffer_cl_, CL_FALSE, 0, sizeof(T), &A, nullptr,
+      queue.enqueueWriteBuffer(buffer_cl_, opencl_context.in_order(), 0,
+                               sizeof(T), &A, nullptr, &transfer_event);
+      this->add_write_event(transfer_event);
+    } catch (const cl::Error& e) {
+      check_opencl_error("matrix constructor", e);
+    }
+  }
+
+  /**
+   * Constructor for the matrix_cl that
+   * creates a copy of a rvalue reference
+   * of a scalar on the OpenCL device.
+   * Regardless of `partial_view`, whole matrix is stored.
+   *
+   * @param A the scalar
+   * @param partial_view which part of the matrix is used
+   */
+  explicit matrix_cl(const T&& A,
```

Comment:
```suggestion
template<typename Scal, typename = require_same<T,std::remove_reference_t<Scal>>>
  explicit matrix_cl(const Scal&& A,
```
I don't think we need two constructors for scalars. Just this one should be enough if you make a small change, so you can delete the one above.

Reply:
Yep, didnt think of this one. Thanks!